torch/_inductor/codegen/triton.py

```diff
@@ -2364,63 +2368,103 @@ def scan(
+                default_tensor = self.cse.generate(
+                    self.body,
+                    f"tl.full({[1] * self.triton_tensor_ndim()}, {default}, {triton_compute_type(dtype)})",
+                )
+                return self.cse.generate(
+                    self.compute, f"tl.where({cond}, {value}, {default_tensor})"
+                )
+
+            value = self.cse.generate(
+                self.compute,
+                f"tl.broadcast_to({value}.to({triton_compute_type(dtype)}), {self.dense_size_str()})",
+            )
+            broadcasted_values.append(value)
+
+            default = triton_constant(init)
             default_tensor = self.cse.generate(
                 self.body,
                 f"tl.full({[1] * self.triton_tensor_ndim()}, {default}, {triton_compute_type(dtype)})",
             )
-            return self.cse.generate(
-                self.compute, f"tl.where({cond}, {value}, {default_tensor})"
-            )
+            dim = len(self.range_trees) - 1 - int(bool(self.no_x_dim))
+            acc_type = triton_acc_type(dtype)
+            cond = " & ".join(masks)
 
-        if self.persistent_reduction:
-            masked_value = where_cond(value)
-            result_var = self.cse.generate(
-                self.compute,
-                f"tl.associative_scan({masked_value}, {dim}, {combine_helper_fn})",
-            )
-        else:
-            accumulator = self.cse.newvar()
-            reduced_size = self.dense_size_list()
-            reduced_size[-1] = "1"
-            reduced_size = f"[{', '.join(reduced_size)}]"
+            if self.persistent_reduction:
+                masked_value = where_cond(value)
+                masked_values.append(masked_value)
+            else:
+                accumulator = self.cse.newvar()
+                reduced_size = self.dense_size_list()
+                reduced_size[-1] = "1"
+                reduced_size = f"[{', '.join(reduced_size)}]"
 
-            self.body.writeline(
-                f"{accumulator} = tl.full({reduced_size}, {default}, {acc_type})"
-            )
+                self.body.writeline(
+                    f"{accumulator} = tl.full({reduced_size}, {default}, {acc_type})"
+                )
 
-            masked_value = where_cond(value)
-            partial_reduce = self.cse.generate(
-                self.compute,
-                self.reduction_resize(
-                    f"tl.reduce({value}, {dim}, {combine_helper_fn})"
-                ),
-            )
-            acc_next = combine_fn(accumulator, partial_reduce)
-            partial_scan = self.cse.generate(
-                self.compute,
-                f"tl.associative_scan({masked_value}, {dim}, {combine_helper_fn})",
+                masked_value = where_cond(value)
+
+                masked_values.append(masked_value)
+                accumulators.append(accumulator)
+
+        def csv(values):
+            return " ".join(f"{value}," for value in values)
```

Comment:
Why the trailing comma?

Reply:
To make it easier to have 1-dimensional tuples.
To get  `a, = x`  for 1-d tuples and `a, b, = x` for 2-d tuples.